builder/dockerfile/builder.go

```diff
@@ -293,21 +351,17 @@ func checkDispatchDockerfile(dockerfile *parser.Node) error {
 	return nil
 }
 
-func dispatchFromDockerfile(b *Builder, result *parser.Result, dispatchState *dispatchState) (*container.Config, error) {
-	shlex := NewShellLex(result.EscapeToken)
+func parseFromDockerfile(b *Builder, result *parser.Result, state *command.ParsingResult) (*command.ParsingResult, error) {
```

Comment:
I think with the changes to `parseDockerfileWithCancellation()` these two functions will be identical, and one can be removed.